src/compiler/checker.ts

```diff
@@ -6652,6 +6652,15 @@ namespace ts {
                     return links.resolvedSymbol = lateSymbol;
                 }
             }
+            else {

```

Comment:
when does this case happen? `lateSymbols` is created in `getResolvedMembersOrExportsOfSymbol`, so I think it can only happen when the same name happens twice (or more) during the same `for (const decl of symbol.declarations)` loop. What is code that would have this structure?

Reply:
Is it two declarations with properties of the same name that should merge?

Reply:
In this case the interface HKT is defined in two files as extension of each other. If hkt is type checked, since that's the first declaration, we will get merged symbol => get resolved members on that symbol and things should work fine when resolving alias since HKT will use merged symbol.
Now in incremental scenarios or editor when hkt is not type checked, while typechecking hkt extended in main.ts, the parent = symbol of declaration of hkt interface in main.ts. While typechecking properties, the lateSymbol is bound and set in resolvedMembers of HKT interface declaration in main.ts. 
But when resolving alias, we get merged symbol and late bound symbol has already a symbol but its not added to the merged symbol's resolvedMembers.